staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -537,11 +537,8 @@ func AddPruningFlags(cmd *cobra.Command, prune *bool, pruneAllowlist *[]string,
 	}
 }
 
-func AddSubresourceFlags(cmd *cobra.Command, subresource *string, usage string, allowedSubresources ...string) {
-	cmd.Flags().StringVar(subresource, "subresource", "", fmt.Sprintf("%s Must be one of %v. This flag is beta and may change in the future.", usage, allowedSubresources))
-	CheckErr(cmd.RegisterFlagCompletionFunc("subresource", func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
```

Comment:
It makes sense to remove the validations but why are we losing completions?. I think, we can still pass the list (by also adding `resize`) to this function.

Reply:
Done.